### ISSUE TITLE
`awsproxy` package should be `awsproxyextension`

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-collector-contrib#30464]
### Component(s)

extension/awsproxy

### Describe the issue you're reporting

This package doesn't follow the loose convention we have set for extensions. With the addition of generated tests, it would make sense to be able to programmatically deduce the package name from type and class.

**Comment by crobert-1** [comment]
I agree that it would be good to be consistent, but I'm seeing quite a few packages that don't include the component type (e.g. `receiver`, `connecter`, etc) appended to their directory. Is there a central tracking issue for this work that we could link all non-conforming components to, and update from there?
